data_steward/cdr_cleaner/cleaning_rules/truncate_rdr_using_date.py

```diff
@@ -0,0 +1,174 @@
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'All rows of data in the August RDR export with dates after 08/01/2020 will be truncated.'
+        super().__init__(issue_numbers=['DC1009'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.RDR],
+                         affected_tables=TRUNCATE_TABLES,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+        counter = 0
+        sandbox_queries = []
+        truncate_queries = []
+        for table in self.affected_tables:
+
+            save_changed_rows = {
+                cdr_consts.QUERY:
+                    SANDBOX_QUERY.render(
+                        project=self.project_id,
+                        dataset=self.dataset_id,
+                        sandbox_dataset=self.sandbox_dataset_id,
+                        intermediary_table=self.get_sandbox_tablenames()
+                        [counter],
+                        table_name=table,
+                        field_name=TABLES_DATES_FIELDS[table],
+                        cutoff_date=CUTOFF_DATE)
+            }
+
+            sandbox_queries.append(save_changed_rows)
+
+            truncate_query = {
+                cdr_consts.QUERY:
+                    TRUNCATE_ROWS.render(project=self.project_id,
+                                         dataset=self.dataset_id,
+                                         table_name=table,
+                                         field_name=TABLES_DATES_FIELDS[table],
+                                         cutoff_date=CUTOFF_DATE),
+            }
+
+            truncate_queries.append(truncate_query)
+            counter += 1
+
+        return sandbox_queries + truncate_queries
+
+    def setup_rule(self, client):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
+
+    def setup_validation(self, client):
+        """
+        Run required steps for validation setup
+        """
+        pass
+
+    def validate_rule(self, client):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+        """
+        pass
```

Comment:
Should we add `raise NotImplementedError("Please fix me.")` to these so that when we do need to implement them they are easily found?